engine/runtime/src/main/scala/org/enso/compiler/SerializationManager.scala

```diff
@@ -120,7 +123,7 @@ final class SerializationManager(
       module.getSource,
       useGlobalCacheLocations
     )
-    if (compiler.context.getEnvironment.isCreateThreadAllowed) {
+    if (useThreadPool) {
```

Comment:
Please use the "thread pool" of [system threads](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/instrumentation/TruffleInstrument.Env.html#createSystemThread-java.lang.Runnable-) everytime! They are available all the time and we don't need to ask the environment if `isCreateThreadAllowed`. It is.

Reply:
Just make sure we always allocate [system thread](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/instrumentation/TruffleInstrument.Env.html#createSystemThread-java.lang.Runnable-) - a thread that can perform various background tasks, but cannot execute Enso code.

Reply:
I checked it. The thread pool uses system threads